plasma_framework/contracts/src/transactions/PaymentTransactionModel.sol

```diff
@@ -35,20 +51,24 @@ library PaymentTransactionModel {
         require(rlpOutputs.length <= MAX_OUTPUT_NUM, "Transaction outputs num exceeds limit");
 
         uint txType = rlpTx[0].toUint();
+        require(txType > 0, "Transaction type must not be 0");
 
         bytes32[] memory inputs = new bytes32[](rlpInputs.length);
         for (uint i = 0; i < rlpInputs.length; i++) {
-            bytes32 input = bytes32(rlpInputs[i].toUint());
+            bytes32 input = rlpInputs[i].toBytes32();
+            // Disallow null inputs
+            require(uint256(input) != 0, "Null input not allowed");
             inputs[i] = input;
         }
 
         PaymentOutputModel.Output[] memory outputs = new PaymentOutputModel.Output[](rlpOutputs.length);
         for (uint i = 0; i < rlpOutputs.length; i++) {
             PaymentOutputModel.Output memory output = PaymentOutputModel.decode(rlpOutputs[i]);
+            require(output.outputType > 0, "Output type must not be 0");
```

Comment:
1.) Why are the checks related to `PaymentOutputModel` not in `decode`?
2.) I think we concluded that the amount has be greater than 0?

Reply:
> 1.) Why are the checks related to PaymentOutputModel not in decode?

Reply:
> 2.) I think we concluded that the amount has be greater than 0?